src/test/java/com/commercetools/sync/shoppinglists/helpers/LineItemReferenceResolverTest.java

```diff
@@ -0,0 +1,158 @@
+package com.commercetools.sync.shoppinglists.helpers;
+
+import com.commercetools.sync.commons.exceptions.ReferenceResolutionException;
+import com.commercetools.sync.services.TypeService;
+import com.commercetools.sync.shoppinglists.ShoppingListSyncOptions;
+import com.commercetools.sync.shoppinglists.ShoppingListSyncOptionsBuilder;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.models.SphereException;
+import io.sphere.sdk.shoppinglists.LineItemDraft;
+import io.sphere.sdk.shoppinglists.LineItemDraftBuilder;
+import io.sphere.sdk.types.CustomFieldsDraft;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.commercetools.sync.commons.MockUtils.getMockTypeService;
+import static com.commercetools.sync.commons.helpers.BaseReferenceResolver.BLANK_KEY_VALUE_ON_RESOURCE_IDENTIFIER;
+import static com.commercetools.sync.commons.helpers.CustomReferenceResolver.TYPE_DOES_NOT_EXIST;
+import static com.commercetools.sync.shoppinglists.helpers.LineItemReferenceResolver.FAILED_TO_RESOLVE_CUSTOM_TYPE;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class LineItemReferenceResolverTest {
+
+    private TypeService typeService;
+
+    private LineItemReferenceResolver referenceResolver;
+
+    /**
+     * Sets up the services and the options needed for reference resolution.
+     */
+    @BeforeEach
+    void setup() {
+        typeService = getMockTypeService();
+
+        final ShoppingListSyncOptions syncOptions = ShoppingListSyncOptionsBuilder.of(mock(SphereClient.class)).build();
+        referenceResolver = new LineItemReferenceResolver(syncOptions, typeService);
+    }
+
+    @Test
+    void resolveReferences_WithNonNullIdOnCustomTypeResId_ShouldResolveCustomTypeReference() {
```

Comment:
Does it resolve anything in this test? I see that `lineItemDraft` is the same as `resolvedDraft`.

Reply:
you are right, check with this name, please: https://github.com/commercetools/commercetools-sync-java/pull/602/commits/197871b54f23763c3218e14ea19e59aba4a88d1c